packages/financial-templates-lib/src/logger/SlackTransport.ts

```diff
@@ -68,20 +91,18 @@ function slackFormatter(info) {
       }
       // If the value in the message is an object then spread each key value pair within the object.
       else if (typeof info[key] === "object" && info[key] !== null) {
-        formattedResponse.blocks.push({ type: "section", text: { type: "mrkdwn", text: ` • _${key}_:\n` } });
+        // Note: create local reference to this object, so we can modify it in the if statement.
+        const newBlock: SectionBlock = { type: "section", text: { type: "mrkdwn", text: ` • _${key}_:\n` } };
+        formattedResponse.blocks.push(newBlock);
```

Comment:
I'm a bit confused by how this works exactly. you are creating a `newBlock` locally here, then assigning stuff to it _after _ having pushed it into the `formattedResponce`. how is this updating the f`formattedResponse`?

Reply:
I looked this up when implementing to verify I was thinking about it correctly. When you push to an array, it pushes a _reference_ to the object to the array. So if you keep the object around and keep modifying it, the object within the array will change too (since they both reference the same underlying data structure). See https://stackoverflow.com/a/8661088.

Reply:
Does that make sense?

Reply:
It does make sense. I hate that that's a thing haha. Really not a fan of how JS treats this differently depending on the type. It might be useful to add a comment about this pushing in a _pointer_ and that means you are modifying the original data.

Reply:
Will add a comment!